src/test/operators/join_test_runner.cpp

```diff
@@ -0,0 +1,590 @@
+    for (const auto join_mode : {JoinMode::AntiNullAsTrue, JoinMode::AntiNullAsFalse}) {
+      for (const auto left_table_size : all_table_sizes) {
+        for (const auto right_table_size : all_table_sizes) {
+          auto join_test_configuration = default_configuration;
+          join_test_configuration.join_mode = join_mode;
+          join_test_configuration.nullable_left = true;
+          join_test_configuration.nullable_right = true;
+          join_test_configuration.input_left.table_size = left_table_size;
+          join_test_configuration.input_right.table_size = right_table_size;
+
+          add_configuration_if_supported(join_test_configuration);
+        }
+      }
+    }
+
+    // JoinOperators need to deal with differently sized Chunks (e.g., smaller last Chunk)
+    // Trigger those via testing all table_size/chunk_size combinations
+    for (const auto& left_table_size : all_table_sizes) {
+      for (const auto& right_table_size : all_table_sizes) {
+        for (const auto& chunk_size : all_chunk_sizes) {
+          auto join_test_configuration = default_configuration;
+          join_test_configuration.input_left.table_size = left_table_size;
+          join_test_configuration.input_right.table_size = right_table_size;
+          join_test_configuration.input_left.chunk_size = chunk_size;
+          join_test_configuration.input_right.chunk_size = chunk_size;
+
+          add_configuration_if_supported(join_test_configuration);
+        }
+      }
+    }
+
+    // Join operators tend to instantiate templates for data types, but also for JoinModes (see probe_semi_anti() in
+    // join_hash_steps.hpp) so combinations of those are issued
+    for (const auto join_mode : all_join_modes) {
+      for (const auto data_type : all_data_types) {
+        auto join_test_configuration = default_configuration;
+        join_test_configuration.data_type_left = data_type;
+        join_test_configuration.data_type_right = data_type;
+        join_test_configuration.join_mode = join_mode;
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // Different JoinModes have different handling of NULL values.
+    // Additionally, JoinOperators (e.g., JoinSortMerge) have vastly different paths for different PredicateConditions
+    // Test all combinations
+    for (const auto& join_mode : all_join_modes) {
+      for (const auto predicate_condition : all_predicate_conditions) {
+        auto join_test_configuration = default_configuration;
+        join_test_configuration.join_mode = join_mode;
+        join_test_configuration.nullable_left = true;
+        join_test_configuration.nullable_right = true;
+        join_test_configuration.predicate_condition = predicate_condition;
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // The input tables are designed to have exclusive values. Test that these are handled correctly for different
```

Comment:
What are exclusive values? Why does swapping (potentially) change the tested behavior?

Reply:
Added `// The input tables are designed to have exclusive values (i.e. values not contained in the other tables).`

Think of Semi/Anti joins. An exclusive value on the left side will not appear in the output of a Semi, but in the output of an Anti. Similarily, an exclusive value on the left side will have to be filled up with NULLs by a LeftOuter join, but discarded by a RightOuter join.